client/keys/utils.go

```diff
@@ -83,9 +90,13 @@ func ReadPassphraseFromStdin(name string) (string, error) {
 }
 
 // initialize a keybase based on the configuration
-func GetKeyBaseFromDir(rootDir string) (keys.Keybase, error) {
+func GetKeyBaseFromDir(rootDir string, readonly bool) (keys.Keybase, error) {
+	var opts *opt.Options
+	if readonly {
+		opts = &opt.Options{ReadOnly: true}
+	}
 	if keybase == nil {
-		db, err := dbm.NewGoLevelDB(KeyDBName, filepath.Join(rootDir, "keys"))
+		db, err := dbm.NewGoLevelDBWithOpts(KeyDBName, filepath.Join(rootDir, "keys"), opts)
```

Comment:
Is it okay to pass `nil` to `NewGoLevelDBWithOpts`?

Reply:
Yes, it falls back to default values.

Reply:
Neat! Thanks.